src/ProjectNeon/Assets/Cards/Effect.cs

```diff
@@ -3,4 +3,6 @@
 [CreateAssetMenu()]
 public class EffectAction : ScriptableObject
 {
+    [SerializeField]
+    private ScriptableObject effect;
```

Comment:
How will this actually work in the code?

If it's just a scriptable object, then we will have to use Reflection or something in order to actually execute it.

This design will probably need to be more like Command Pattern, where each effect is a Command.

Reply:
@SilasReinagel Actually I didn't understood quite well `EffectAction` class significance into the project. The concept is clear for me, just its implementation as a class makes me puzzled. The `effect` variable exists here just due to the fact that I wasn't able to extend / implement `EffectAction` class in a polymorphic way using UnityEditor; I wasn't even able to add scripts to it using just the UI, so I've created this field so I could put scripts into `EffectAction`. I don't know how some concepts are transposed to Unity universe, but in 'traditional OO' I'd create an Interface `EffectAction` and implement it as many times as desired for each effect we have. Wouldn't be better if we had `EffectAction` as an interface and follow this proposed logic?